pytorch_lightning/loops/epoch/evaluation_epoch_loop.py

```diff
@@ -88,6 +88,21 @@ def on_run_start(  # type: ignore[override]
         # add the previous `fetched` value to properly track `is_last_batch` with no prefetching
         data_fetcher.fetched += self.batch_progress.current.ready
 
+        dataloader_idx = f"_{kwargs['dataloader_idx']}" if "dataloader_idx" in kwargs else ""
+        if self.trainer.testing:
+            suffix = f"test_dataloader{dataloader_idx}_next"
+        else:
+            suffix = f"val_dataloader{dataloader_idx}_next"
+
+        def _on_before_fetch() -> None:
+            self.trainer.profiler.start(f"[{self.__class__.__name__}].{suffix}")
+
+        def _on_after_fetch() -> None:
+            self.trainer.profiler.stop(f"[{self.__class__.__name__}].{suffix}")
```

Comment:
I'm uncomfortable using local functions as they will not work if this class gets pickled.

Reply:
Same here. Let's avoid them.